src/parser/smt2/smt2.cpp

```diff
@@ -1038,48 +1039,95 @@ cvc5::Term Smt2::applyParseOp(ParseOp& p, std::vector<cvc5::Term>& args)
   if (p.d_kind == cvc5::UNDEFINED_KIND && isIndexedOperatorEnabled(p.d_name))
   {
     // Resolve indexed symbols that cannot be resolved without knowing the type
-    // of the arguments. This is currently limited to `to_fp`.
-    Assert(p.d_name == "to_fp");
+    // of the arguments. This is currently limited to `to_fp`, `tuple.select`,
+    // and `tuple.update`.
     size_t nchildren = args.size();
-    if (nchildren == 1)
+    if (p.d_name == "to_fp")
     {
-      kind = cvc5::FLOATINGPOINT_TO_FP_FROM_IEEE_BV;
-      op = d_solver->mkOp(kind, p.d_indices);
-    }
-    else if (nchildren > 2)
-    {
-      std::stringstream ss;
-      ss << "Wrong number of arguments for indexed operator to_fp, expected "
-            "1 or 2, got "
-         << nchildren;
-      parseError(ss.str());
-    }
-    else if (!args[0].getSort().isRoundingMode())
-    {
-      std::stringstream ss;
-      ss << "Expected a rounding mode as the first argument, got "
-         << args[0].getSort();
-      parseError(ss.str());
+      if (nchildren == 1)
+      {
+        kind = cvc5::FLOATINGPOINT_TO_FP_FROM_IEEE_BV;
+        op = d_solver->mkOp(kind, p.d_indices);
+      }
+      else if (nchildren > 2)
+      {
+        std::stringstream ss;
+        ss << "Wrong number of arguments for indexed operator to_fp, expected "
+              "1 or 2, got "
+           << nchildren;
+        parseError(ss.str());
+      }
+      else if (!args[0].getSort().isRoundingMode())
+      {
+        std::stringstream ss;
+        ss << "Expected a rounding mode as the first argument, got "
+           << args[0].getSort();
+        parseError(ss.str());
+      }
+      else
+      {
+        cvc5::Sort t = args[1].getSort();
+
+        if (t.isFloatingPoint())
+        {
+          kind = cvc5::FLOATINGPOINT_TO_FP_FROM_FP;
+          op = d_solver->mkOp(kind, p.d_indices);
+        }
+        else if (t.isInteger() || t.isReal())
+        {
+          kind = cvc5::FLOATINGPOINT_TO_FP_FROM_REAL;
+          op = d_solver->mkOp(kind, p.d_indices);
+        }
+        else
+        {
+          kind = cvc5::FLOATINGPOINT_TO_FP_FROM_SBV;
+          op = d_solver->mkOp(kind, p.d_indices);
+        }
+      }
     }
-    else
+    else if (p.d_name == "tuple.select" || p.d_name == "tuple.update")
     {
-      cvc5::Sort t = args[1].getSort();
-
-      if (t.isFloatingPoint())
+      bool isSelect = (p.d_name == "tuple.select");
+      if (p.d_indices.size() != 1)
```

Comment:
Here, we have (tuple.project) for multiple indices for tuple.select. Do we have an update operator with multiple indices.

Reply:
`tuple.select` takes one argument only.  

In terms of the relationship to `tuple.project`, I think it might make sense for consistency to have a TUPLE_SELECTOR operator on the API level.  This would eliminate the need for this delayed resolution altogether.